vita3k/gui/src/settings_dialog.cpp

```diff
@@ -94,18 +97,20 @@ void draw_settings_dialog(GuiState &gui, HostState &host) {
             if (ImGui::IsItemHovered())
                 ImGui::SetTooltip("Select your desired modules.");
             ImGui::PushItemWidth(240);
-            static std::vector<bool> module_select(gui.modules_list.size());
             if (ImGui::ListBoxHeader("##modules_list", static_cast<int>(gui.modules_list.size()), 6)) {
                 for (int m = 0; m < gui.modules_list.size(); m++) {
-                    auto modules = std::find(host.cfg.lle_modules.begin(), host.cfg.lle_modules.end(), gui.modules_list[m]);
-                    if (modules != host.cfg.lle_modules.end() && !module_select[m])
-                        module_select[m] = true;
-                    if (!gui.module_search_bar.PassFilter(gui.modules_list[m].c_str()))
-                        continue;
-                    if (ImGui::Selectable(gui.modules_list[m].c_str(), module_select[m]) && modules == host.cfg.lle_modules.end())
-                        host.cfg.lle_modules.push_back(gui.modules_list[m]);
-                    else if (!module_select[m] && modules != host.cfg.lle_modules.end())
-                        host.cfg.lle_modules.erase(modules);
+                    const auto module = std::find(host.cfg.lle_modules.begin(), host.cfg.lle_modules.end(), gui.modules_list[m]);
+                    const bool module_existed = (module != host.cfg.lle_modules.end());
+                    if (module != host.cfg.lle_modules.end() && !gui.lle_modules_selected[m])
+                        if (!gui.module_search_bar.PassFilter(gui.modules_list[m].c_str()))
+                            continue;
```

Comment:
remove first line, that broken seach
```suggestion
                    if (!gui.module_search_bar.PassFilter(gui.modules_list[m].c_str()))
                        continue;
```